src/lib/io/sol-iio.c

```diff
@@ -1762,3 +1764,108 @@ sol_iio_mount_calibration(struct sol_iio_device *device, sol_direction_vector *v
     SOL_DBG("%f-%f-%f", value->x, value->y, value->z);
     return 0;
 }
+
+SOL_API char *
+sol_iio_channel_get_name(struct sol_iio_channel *channel)
+{
+    SOL_NULL_CHECK(channel, NULL);
+
+    return channel->name;
+}
+
+SOL_API int
+sol_iio_device_get_scale(struct sol_iio_device *device, const char *prefix_name, double *scale)
+{
+    char path[PATH_MAX];
+    char *pure_name = NULL;
+    int r;
+
+    SOL_NULL_CHECK(device, -EINVAL);
+    SOL_NULL_CHECK(prefix_name, -EINVAL);
+    SOL_NULL_CHECK(scale, -EINVAL);
+
+#define GET_SCALE(_name) \
+    do { \
+        if (craft_filename_path(path, sizeof(path), CHANNEL_SCALE_PATH, \
+            device->device_id, _name)) { \
+            r = sol_util_read_file(path, "%lf", scale); \
+            goto end; \
+        } \
+    } while (0)
+
+    GET_SCALE(prefix_name);
+    pure_name = channel_get_pure_name(prefix_name);
+    GET_SCALE(pure_name);
+
```

Comment:
this code is almost the same of `channel_get_scale()`. Wouldn't be better both use same code ? You can put this code in another function and both call it. 

One more thing, the scale is not associated with a channel ? if yes, is it possible get it directly from the desired channel looking the channel in the channels vector ?